### PR TITLE
[READY] Adds Clarke, your new mech buddy!

### LINKED ISSUE #39294: Bug during Firefigher APLU construction
Constructing a Ripley Firefighter mech has two steps that add armor: Internal and external. When applying external armor, clicking the mech with plasteel for the first time consumes 5 plasteel, but shows no indicator for it. 
Clicking the mech a second time consumes another 5 plasteel and properly finishes the step with a "You install the external reinforced armor layer to the Firefighter chassis." message.

Intended behavior would be that the "install external armor" step is either completed in one click or gives an indicator that you have to click it with plasteel for a second time.

### PR DESCRIPTION
**Pull request by Fikou**
big thanks to /vg/station coders and spriters Shadowmech88, RealestEstate and ValkyrieSkies
## About The Pull Request
Adds the new Clarke mech, who is a replacement to the firefighter mech
Clarke is made with roundstart materials + 5 gold, which requires miners to work a little before it can be created
clark has lavaproof treads and ash storm protection, and is also really fast, but consumes more energy per step because of that and cannot strafe since he uses treads
clarke's armor is weaker than other mechs
also moves a bunch of ripley specific stuff to general mecha/working
clarke is unlocked with the basic engineering tech node
ripley mk2 now has stronger armor to make it more of a replacement of a hulking mech like firefighter was, but it isnt fireproof
misc changes:
you can now salvage phazon parts
conveyor belts are now in protolathe electronics and are needed as clarke's legs
theres now a clarke action figure
closes #39294

clarke look:
image

## Why It's Good For The Game

firefighter is kind of lame, its just a better ripley and literally a recolor, clarke gives more of a difference, with a high speed and very low defense, also more visual variety

## Changelog
:cl: Fikou (thanks to Shadowmech88 for idea and RealestEstate/ValkyrieSkies for sprites)
add: Clarke Mech!
fix: You can now salvage Phazon parts
del: Firefighter Mech, rest in peace
balance: Ripley MK-II now has similar armor values to firefighter, but without the fireproof stuff
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
Firefighter mech has some uses outside of mining (like its namesake), dont just remove it because another mech replaces one of its uses.

**Comment by HomocidlePancakes** [comment]
Ripley can extinguish fires too ¯\_(ツ)_/¯

**Comment by zxaber** [comment]
But muh APLU MK I-III lore.

Firefighter is a good mech, don't remove it. This mech has its own niche and doesn't need to replace anyone.

**Comment by HomocidlePancakes** [comment]
>Yeah, lets send the not-fireproof mech to fight fires.

All mechs except for MK1 are fireproof. What are you talking about?

**Comment by zxaber** [comment]
That was a bug that I inadvertently caused with the MK-I Ripley rework, and fixed ages ago.

**Comment by TheVekter** [comment]
FUCK YOU BEAT ME TO IT I WAS GONNA DO SOMETHING SIMILAR

Whatever, I'll just add upgrades and shit later. 10/10 good shit.

**Comment by zxaber** [comment]
By the by;
>clarke's armor is really weak, worse than other mechs, even a few hits from a watcher can wreck one

Mining mechs have no offensive weapons to speak of (it takes 30 cutter shots to kill a goliath). Aside from lava walking and a bit of armor, the Firefighter brings nothing to the table for miners. It's slower, and can't fight back against anything that threatens it, and the only saving grace is that it can take a bit of a beating while trying to find an exit from a fight.

The Clarke seems like a neat mech, but if you make it made of paper no one will want to use it.

**Comment by bluetshirtguy** [comment]
Miners don't use mechs not even the 'miner' ripely,focus on making clarke useful on the station not lavaland.

**Comment by GunHog** [comment]
> By the by;
> 
> > clarke's armor is really weak, worse than other mechs, even a few hits from a watcher can wreck one
> 
> Mining mechs have no offensive weapons to speak of (it takes 30 cutter shots to kill a goliath). Aside from lava walking and a bit of armor, the Firefighter brings nothing to the table for miners. It's slower, and can't fight back against anything that threatens it, and the only saving grace is that it can take a bit of a beating while trying to find an exit from a fight.
> 
> The Clarke seems like a neat mech, but if you make it made of paper no one will want to use it.

This. Miners do not take mechs because they are inferior. The plasma cutter used to be decent before the nerf, but now it is only good for mining, not combat. It needs to be good at both to be worth choosing the mech over going on foot.

**Comment by zxaber** [comment]
I think there is certainly potential in making Clarke a mining mech. It certainly looks more at home on Lavaland than on the station with those tracks.

I don't think it should pretend to be a Firefighter replacement, though.

**Comment by bluetshirtguy** [comment]
> I think there is certainly potential in making Clarke a mining mech. It certainly looks more at home on Lavaland than on the station with those tracks.
> 
> I don't think it should pretend to be a Firefighter replacement, though.

Unless its absolutely absurd at mining early game, it isnt of use to the average miner. mechs of any kind is something a new player miner would use. Clarke has far too weak armor and to fighting power and mobility isnt an issue not even over lava that's what rcds are for.
In conclusion miners dont use mechs in the first place. not the firegither, not a gygax/durand not clarke.

**Comment by ATH1909** [comment]
When are we gonna unnerf plasma cutter bolts, but only for mechs? The turrets used by the auxiliary base already fire a special (buffed) type of plasma cutter bolt; why can't mechs?

**Comment by Dennok** [comment]
> When are we gonna unnerf plasma cutter bolts, but only for mechs? The turrets used by the auxiliary base already fire a special (buffed) type of plasma cutter bolt; why can't mechs?

Yes. Plasma Cutters nerfed because you can take 2 and fire both at the same time.

File: code/game/mecha/working/clarke.dm [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	name = "\improper Clarke"
+	icon_state = "clarke"
+	max_temperature = 65000
+	max_integrity = 250
+	step_in = 1.5
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	lights_power = 7
+	deflect_chance = 10
+	step_energy_drain = 15 //slightly higher energy drain since you movin those wheels FAST
+	armor = list("melee" = 20, "bullet" = 10, "laser" = 30, "energy" = 30, "bomb" = 60, "bio" = 0, "rad" = 70, "fire" = 100, "acid" = 100) //low bullet/melee armor to compensate for fire protection and speed
```
Comment (Tlaltecuhtli) [inline code/game/mecha/working/clarke.dm]: why 30 laser armor thats gygax tier of armor

**Comment by Tlaltecuhtli** [comment]
conveyor belt is off topic , same for the durand bounty, if you are doing random stuff put fluid pipes in medlathe

**Comment by Tlaltecuhtli** [comment]
> > When are we gonna unnerf plasma cutter bolts, but only for mechs? The turrets used by the auxiliary base already fire a special (buffed) type of plasma cutter bolt; why can't mechs?
> 
> Yes. Plasma Cutters nerfed because you can take 2 and fire both at the same time.

plasma cutters were nerfed because they were almost free and  mass printed during blob for pka damage once the  blob inevitably breaks a wall and the zone gets depressurized

File: code/game/mecha/working/clarke.dm (hunk 1 above)
Comment (HomocidlePancakes) [reply code/game/mecha/working/clarke.dm]: What are you on, Tlal? FF has the same laser armor.

**Comment by Tharcoonvagh** [comment]
Dont remove my firefighter you monster

**Comment by zxaber** [comment]
Handheld cutter nerf also nerfed the mech cutters, which didn't make a lot of sense as none of the given reasons for the nerf applied to mech cutters.

That being said, a revert specifically for mech cutters has been tried twice and been shot down both times. They are pretty much stuck dealing minimal damage and that's just how it is.

File: code/game/mecha/working/clarke.dm (hunk 1 above)
Comment (Tlaltecuhtli) [reply code/game/mecha/working/clarke.dm]: ff doesnt have wheels and op said "light armor high speed"

**Comment by TheVekter** [comment]
> odysseus is now unlocked with basic biotech node instead of roundstart
> you can now salvage phazon parts
> some of the mecha bounty prices were risen

These probably need to be atomized into another PR.

**Comment by Tlaltecuhtli** [comment]
no sprite for it and what would a medical mech have to do with mining

**Comment by Dennok** [comment]
<details>
  <summary>some icons</summary>

old

new


old

new


old

new


old

new
</details>

**Comment by Dennok** [comment]
Dont remove firefighter. Just add new mecha.

**Comment by Fikou** [comment]
i made the mk2 ripley have more similar armor stats to the firefighter to act more as a replacement if people used firefighters to be a big hulking mecha with nice armor

**Comment by TheVekter** [comment]
> i made the mk2 ripley have more similar armor stats to the firefighter to act more as a replacement if people used firefighters to be a big hulking mecha with nice armor

That doesn't really address my problem. Why would I ever take a mech out to mine if it dies at the first sign of fauna and can't be upgraded with Goliath panels?

**Comment by Fikou** [comment]
> That doesn't really address my problem. Why would I ever take a mech out to mine if it dies at the first sign of fauna and can't be upgraded with Goliath panels?

its not a combat mech and without oranges changing his opinion on giving mechs actual kas or something it wont be a combat mech
the mech can run around really fast on lavaland mining allowing you to get out of mob reach while in something that cant be stunned or disoriented unless under extensive damage
the mech can mine and move fast and moves on lava and during ash storms, letting you explore easier

**Comment by zxaber** [comment]
I've slept on it, and I've come to the opinion that you should remove the ability for Clarke to carry cargo at all. Cargo carrying is pretty much the default purpose of the APLU series. You say that this mech makes the Firefighter redundant, but in reality Clarke actually does everything the MK-I Ripley can do.

MK-I was designed to allow for a faster speed at the expense of having no atmos protection and no real protection from projectiles. Not only does Clarke move even faster than the Ripley, but it shares none of the downsides at all.

Instead of the ability to load items, give Clarke a built-in ore box and a button in the View Info window to empty the box.

**Comment by FantasticFwoosh** [comment]
I have a lot of mixed feelings, its gorgeous but also makes existing mechs look like trash, whilst being mechanically different. Atmos & Engineering can use it but it doesn't solve any problems about what it is meant to be replacing for the niche of the Firefighter Ripley, and therefore shouldnt be considered a replacement but a addition because it squarely can't fufill the role.

Outside of the scope of this PR, conveyor belt legs for lava immunity should be availible for all mechs as a modified specification, and to change the structural format of mechs to having a deeper mechanical purpose. Angry mech tank Gigaxes being able to roll over people Mulebot style with the rush ability etc.

**Comment by Tlaltecuhtli** [comment]
make clarke for firefighting  and ripley mk2 for mining

**Comment by Tharcoonvagh** [comment]
>miners dont take mechs

Roboticists do, mostly when miners are dead/absent

**Comment by Fikou** [comment]
> I have a lot of mixed feelings, its gorgeous but also makes existing mechs look like trash

not stopping you from remaking their sprites 😏

**Comment by OverlyGeneric** [comment]
sorry, you've made **_quality-content_**.

that's illegal, sir.

image

**Comment by PKPenguin321** [comment]
The north sprite is shorter than the rest

**Comment by Fikou** [comment]
> The north sprite is shorter than the rest

but it isnt

**Comment by PKPenguin321** [comment]
Capture+_2020-02-26-12-48-39
wtf

**Comment by Fikou** [comment]
> Capture+_2020-02-26-12-48-39
> wtf

look at the sprite left to it

**Comment by PKPenguin321** [comment]
Oh that's the ejected sprite nvm

**Comment by Fikou** [comment]
i now have image proof that pkp is blind or crazy

**Comment by TheMidnightRose** [comment]
As someone who used mining mechs before and after the change to cutters, miners will likely ignore the mech with its paper thin armor. However if you allow us to add goliath plates to improve the armor, with each plate increasing it slightly more than when you add them to APLU series mechs then we might consider using the Clarke. 
To fix the issue with cargo capacity, just make the ONLY thing it can hold being a ore crate, with some message being like 'The (object) you lifted does not seem to fit in the ore crate hold.' and for smaller items 'The (object) in the ore crate hold falls out, unable to be held in the hold.'
If you want to improve its ability to fight on lavaland, give it a proto-kinetic accelerator addon that can be adjusted by miners on the fly with modkits they use for the base gun with one or two more slots for modkits. Though make it unable to use the PKA on station unless the miners modify it with a standard firing pin from security.

**Comment by Fikou** [comment]
>However if you allow us to add goliath plates to improve the armor, with each plate increasing it slightly more than when you add them to APLU series mechs then we might consider using the Clarke.

sprite it and i will
> To fix the issue with cargo capacity, just make the ONLY thing it can hold being a ore crate, with some message being like 'The (object) you lifted does not seem to fit in the ore crate hold.' and for smaller items 'The (object) in the ore crate hold falls out, unable to be held in the hold.'

already done
> If you want to improve its ability to fight on lavaland, give it a proto-kinetic accelerator addon that can be adjusted by miners on the fly with modkits they use for the base gun with one or two more slots for modkits. Though make it unable to use the PKA on station unless the miners modify it with a standard firing pin from security.

as i already said, oranges doesnt like working mechs with good guns

**Comment by Pumpkinoe** [comment]
is it spaceproof

**Comment by WarlockD** [comment]
Eh, it just makes the firefighter..er..the clark more of a rescue-in-lavaland mech and the ripley 2 for everything else.  I don't really see a problem.  I still don't know why a proto wouldn't be allowed for mechs in lavaland though.

**Comment by CuteTuret** [comment]
Hmm, imagine if it had a built-in custom drill arm that was stronger than the diamond drill (not for walls though)

**Comment by Fikou** [comment]
i guess this doesnt fit in with oranges' new vision for mechs
ill leave it open just in case some epic maintainer would like to merge it while we are waiting for actual lavaland mechs ¯\_(ツ)_/¯

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -3,7 +3,7 @@
 	desc = "Exosuit"
 	icon = 'icons/mecha/mecha.dmi'
 	density = TRUE //Dense. To raise the heat.
-	opacity = 1 ///opaque. Menacing.
+	opacity = 1 //opaque. Menacing.
```
Comment (Rohesie) [inline code/game/mecha/mecha.dm]:
```suggestion
	opacity = TRUE //opaque. Menacing.
```

File: code/game/mecha/working/clarke.dm [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	wreckage = /obj/structure/mecha_wreckage/clarke
+	enter_delay = 40
+	canstrafe = FALSE
+	/// Handles an internal ore box for Clarke
+	var/obj/structure/ore_box/box
+
+/obj/mecha/working/clarke/Initialize()
+	. = ..()
+	box = new /obj/structure/ore_box(src)
+	var/obj/item/mecha_parts/mecha_equipment/ME = new /obj/item/mecha_parts/mecha_equipment/orebox_manager(src)
```
Comment (Rohesie) [inline code/game/mecha/working/clarke.dm]:
```suggestion
	var/obj/item/mecha_parts/mecha_equipment/orebox_manager/ME = new(src)
```

File: code/game/mecha/working/clarke.dm [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	if(.)
+		var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]
+		hud.add_hud_to(H)
+
+/obj/mecha/working/clarke/go_out()
+	if(isliving(occupant))
+		var/mob/living/L = occupant
+		var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]
+		hud.remove_hud_from(L)
+	..()
```
Comment (Rohesie) [inline code/game/mecha/working/clarke.dm]:
```suggestion
	return ..()
```

File: code/game/mecha/working/clarke.dm [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	selectable = FALSE
+	detachable = FALSE
+	salvageable = FALSE
+	/// Var to avoid istype checking every time the topic button is pressed. This will only work inside Clarke mechs.
+	var/obj/mecha/working/clarke/hostmech
+
+/obj/item/mecha_parts/mecha_equipment/orebox_manager/attach(obj/mecha/M)
+	if(istype(M, /obj/mecha/working/clarke))
+		hostmech = M
+	. = ..()
```
Comment (Rohesie) [inline code/game/mecha/working/clarke.dm]:
```suggestion
	return ..()
```

File: code/game/mecha/working/clarke.dm [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+/obj/item/mecha_parts/mecha_equipment/orebox_manager/detach()
+	hostmech = null //just in case
+	. = ..()
```
Comment (Rohesie) [inline code/game/mecha/working/clarke.dm]:
```suggestion
	return ..()
```

File: code/game/mecha/working/clarke.dm [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+/obj/item/mecha_parts/mecha_equipment/orebox_manager/Topic(href,href_list)
+	..()
```
Comment (Rohesie) [inline code/game/mecha/working/clarke.dm]:
```suggestion
	. = ..()
```

File: code/game/mecha/working/clarke.dm [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+	if(!hostmech || !hostmech.box)
+		return
+	hostmech.box.dump_box_contents()
+
+/obj/item/mecha_parts/mecha_equipment/orebox_manager/get_equip_info()
+	return "[..()] [hostmech?.box?"<a href='?src=[REF(src)];mode=0'>Unload Cargo</a>":"Error"]"
```
Comment (Rohesie) [inline code/game/mecha/working/clarke.dm]:
```suggestion
	return "[..()] [hostmech?.box ? "<a href='?src=[REF(src)];mode=0'>Unload Cargo</a>" : "Error"]"
```

File: code/game/mecha/working/ripley.dm [hunk 9]
```diff
@@ -220,3 +185,19 @@
...
+  *
+  * Checks for Lavaland pressure, if that works out the mech's speed is equal to fast_pressure_step_in and the cooldown for the mecha drill is halved. If not it uses slow_pressure_step_in and drill cooldown is normal.
+  */
+/obj/mecha/working/ripley/proc/update_pressure()
+	var/turf/T = get_turf(loc)
+
+	if(lavaland_equipment_pressure_check(T))
+		step_in = fast_pressure_step_in
+		for(var/obj/item/mecha_parts/mecha_equipment/drill/drill in equipment)
+			drill.equip_cooldown = initial(drill.equip_cooldown)/2
```
Comment (Rohesie) [inline code/game/mecha/working/ripley.dm]:
```suggestion
			drill.equip_cooldown = initial(drill.equip_cooldown) * 0.5

```

**Pull request merged**